File: Python/AdvancedPython/OperatingSystems/schedulingalgorithms/DiskScheduling/FCFS.py
from matplotlib import pyplot as plt


def fcfsDiskScheduling(sectors: list[int], head):
    seek_time = 0
    x = [head]

    for i in range(len(sectors)):
        seek_time += abs(head - sectors[i])
        head = sectors[i]

    y = [-i for i in range(len(sectors) + 1)]

    plt.rcParams['xtick.top'] = plt.rcParams['xtick.labeltop'] = True

    plt.plot(x + sectors, y, color="green",
             markerfacecolor="blue",
             marker="o",
             markersize=5,
             linewidth=2,
             label="SCAN")

    plt.xlim(0, 200)
    plt.yticks([])
    plt.title('FCFS')
    plt.show()
    return seek_time


if __name__ == '__main__':
    sectors = [176, 79, 34, 60, 92, 11, 41, 114]
    head = 50
    print(fcfsDiskScheduling(sectors, head))

File: CPP/OperatingSystems/SchedulingAlgorithms/DiskScheduling/fcfs.cpp
#include <iostream>
#include <vector>

std:: vector<int> fcfsDiskScheduling(std:: vector<int> sectors, int head)
{
   std:: vector<int> trackSequence;
   int seek_time = 0;

   for(int x: sectors) 
   {
      trackSequence.push_back(std::abs(head - x));
      seek_time += std:: abs(head - x);
      head = x;
   }

   std:: cout << seek_time << std:: endl;

   return trackSequence;
}

int main() 
{
   std:: vector<int> sectors = {176, 79, 34, 60, 92, 11, 41, 114};
   int head = 50;

   std:: vector<int> trackSequence = fcfsDiskScheduling(sectors, head);

   for (int x: trackSequence) 
   {
      std:: cout << x << " ";
   } 
   
   std:: cout << "\n";
}